### ISSUE TITLE
Enhancement: Support `${configDir}` in `tsconfig.json`

### ISSUE THREAD

**Issue by silverwind** [typescript-eslint/typescript-eslint#8992]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

https://github.com/microsoft/TypeScript/issues/57485 added support for `${configDir}` in options like `include` and `exclude`, but it appears typescript-eslint does not support this yet and tryin to set `include: ['${configDir}/**/*']` in `tsconfig.json` results in it no longer finding any files.

**Comment by JoshuaKGoldberg** [comment]
> typescript-eslint errors immediately as it can no longer finding any files to lint.

Not that I don't believe you, but reproduction please. 🙂 https://antfu.me/posts/why-reproductions-are-required

**Comment by silverwind** [comment]
Can't provide a reproduction because Typescript 5.5 is not released yet which is required for this feature.

**Comment by silverwind** [comment]
Actually, maybe it will work with [5.5.0-dev.20240425](https://www.npmjs.com/package/typescript/v/5.5.0-dev.20240425) tag.

**Comment by bradzacher** [comment]
https://github.com/microsoft/TypeScript/pull/58042

This feature will land in TS 5.5
We do not support nightly, beta or RC versions of TS - only official releases.

**Comment by silverwind** [comment]
Yes, but wouldn't you like to get started now? There's still two months until 5.5 release, so plenty of time to get the feature support ready.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/users/dependency-versions#typescript

We do not support the early releases nor do we accept issues against them. This has been our policy for the entire lifetime of the project.

We also do not seek to build early support either as what is in the nighties and betas can change substantially before even the RC release.

Please be patient.
